src/Iam.php

```diff
@@ -347,7 +354,14 @@ protected function loadToken()
     {
         if ($this->iam_token)
         {
-            if ($this->expires_at > time())
+            if ($this->refresh_at < microtime(true)){
+                try {
+                    return $this->newToken();
+                } catch (\Exception $e){
+                    return $this->iam_token;
+                }
+            }
+            else if ($this->expires_at > time())
             {
                 return $this->iam_token;
```

Comment:
This branch will work never, because refresh time always smaller then expires_at.